nil-docs/zkllvm/getting-started/compiling-a-circuit.mdx

```diff
@@ -0,0 +1,193 @@
+import Tabs from '@theme/Tabs'
+import TabItem from '@theme/TabItem'
+
+# Compiling a circuit
+
+This tutorial contains step-by-step instructions on how to compile the [**arithmetic circuit created previously**](./writing-a-simple-circuit).
+
+There are two possible ways for transforming a circuit into a proof binary and Solidity code.
+
+- By calling individual tools
+- By using [**the zkllvm-template project**](https://github.com/NilFoundation/zkllvm-template)
+
+## Calling individual tools
+
+The following diagram shows the flow for converting a `.cpp` or a `.rs` file into a proof binary and Solidity code for deployment.
+
+```mermaid
+%%{
+  init: {
+    'theme': 'base',
+    'themeVariables': {
+      'primaryColor': '#0f0f0f',
+      'primaryTextColor': '#f1f1f1',
+      'primaryBorderColor': '#f1f1f1',
+      'lineColor': '#87B6FC',
+      'secondaryColor': '#f1f1f1',
+      'tertiaryColor': '#2f2f2f',
+    },
+    'flowchart':
+    {
+        'defaultRenderer': 'elk',
+        'curve': 'step'
+    }
+  }
+}%%
+flowchart TB
+    classDef transpilerTitle margin-right: 250px;
+    Transpiler:::transpilerTitle
+
+    CPP(C++ code)
+    RUST(Rust code)
+    CLANG(<code>clang</code>)
+    RUSTCOMP(<code>cargo</code> or <code>rustc</code>)
```

Comment:
Cool diagram!
I think "`cargo` or `rustc`" is a bit inaccurate, because its actually not `cargo` OR `rustc` - its always `rustc` and `cargo` is just a "wapper". Maybe to match the C++ branch and at the same time mention Cargo we can say "`rustc` (`cargo`)" instead.

Reply:
no worries, fixed